Source/ACE/Entity/Player.cs

```diff
@@ -1419,54 +1351,327 @@ private void SendSelf()
+                    var buyrudemsg = new GameMessageSystemChat(rudecomment, ChatMessageType.Tell);
+                    this.Session.Network.EnqueueSend(buyrudemsg);
+                }
             }
 
-            DelayedUpdatePosition(newPosition);
+            yield break;
         }
 
-        public void UpdatePosition(Position newPosition)
+        public IEnumerator ActQueryHealth(ObjectGuid id)
         {
-            Location = newPosition;
-            // character.SetCharacterPosition(newPosition);
-            SendUpdatePosition();
+            WorldObject wo = LandblockManager.GetWorldObject(Session, id);
+            Creature co = wo as Creature;
+            if (co == null)
+            {
+                yield break;
+            }
+
+            float healthPct = 0;
+            lock (co)
+            {
+                healthPct = (float)co.Health.Current / (float)co.Health.MaxValue;
+            }
+
+            Session.Network.EnqueueSend(new GameEventUpdateHealth(Session, id.Full, healthPct));
+
+            yield break;
+        }
+
+        public IEnumerator ActUse(ObjectGuid usableId)
+        {
+            var usableWo = LandblockManager.GetWorldObject(Session, usableId);
+
+            // Move to the object
+            yield return ObjectAction.CallCoroutine(ActMoveTo(usableWo));
+
+            // FIXME(ddevec): Special case for portals :(  Perhaps Usable should be an interface, not base class?
+            var portal = usableWo as Portal;
+            var uo = usableWo as UsableObject;
+
+            // Then use the object
+            // FIXME(ddevec): Sanity checks -- object usable, still close, etc etc
+            lock(usableWo)
+            {
+                if (portal != null)
+                {
+                    portal.OnCollide(this);
+                }
+                else if (uo != null)
+                {
+
+                    // Use it -- let it do the rest
+                    // NOTE(ddevec): We cannot yield in OnUse -- we are holding the lock to that object
+                    //    If we need a back and forth interaction it will have to be via message passing/waiting of some kind
+                    uo.OnUse(this);
+                }
+            }
+            yield break;
+        }
+
+        public IEnumerator ActDrop(ObjectGuid itemGuid)
+        {
+            // FIXME: The object is in your pack not the world!  
+            //WorldObject wo = LandblockManager.GetWorldObject(Session, itemGuid);
+            WorldObject wo = GetInventoryItem(itemGuid);
+
+            if (wo == null)
+            {
+                yield break;
+            }
+
+            lock (wo)
+            {
+                NotifyAndDropItem(wo.Guid);
+            }
+            yield break;
+        }
+
+        public IEnumerator ActMoveTo(ObjectGuid itemGuid)
+        {
+            var item = LandblockManager.GetWorldObject(Session, itemGuid);
```

Comment:
LandblockManager.GetWorldObject is either going to be deprecated or modified to return a new readonly object.  locking the result of this is not viable.